test/e2e: Cleanup net_policies tests

sayboras opened the pull request:
### Description

Please refer to individual commit for more details.

sayboras committed:
test/e2e: Remove matchExpression tests

This commit is to remove matchExpression related tests, as these are
covered in connectivity tests:

https://github.com/cilium/cilium-cli/tree/master/connectivity/manifests

- client-egress-to-entities-world.yaml
- client-egress-to-fqdns-one-one-one-one.yaml
- client-egress-to-echo.yaml
- client-egress-to-echo-expression.yaml
- client-egress-to-echo-expression-deny.yaml

Relates: https://github.com/cilium/cilium-cli/pull/1028
Relates: https://github.com/cilium/cilium/pull/21112
Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
test/e2e: Remove service account related tests

This commit is to remove service account related tests, as these are
covered in connectivity tests:

https://github.com/cilium/cilium-cli/tree/master/connectivity/manifests

- client-egress-to-echo-service-account.yaml
- client-egress-to-echo-service-account-deny.yaml

Relates: https://github.com/cilium/cilium-cli/pull/1028
Relates: https://github.com/cilium/cilium/pull/21112
Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
test/e2e: Remove ICMP related tests

This commit is to remove ICMP related tests, as these are
covered in connectivity tests:

https://github.com/cilium/cilium-cli/tree/master/connectivity/manifests

- echo-ingress-icmp.yaml
- echo-ingress-icmp-deny.yaml

Relates: https://github.com/cilium/cilium-cli/pull/1028
Relates: https://github.com/cilium/cilium/pull/21112
Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
test/e2e: Remove named port related tests

This commit is to remove named port related tests, as these are
covered in connectivity tests:

https://github.com/cilium/cilium-cli/tree/master/connectivity/manifests

- client-egress-l7-http-named-port.yaml
- client-egress-to-echo-named-port-deny.yaml
- echo-ingress-l7-http-named-port.yaml

Relates: https://github.com/cilium/cilium-cli/pull/1028
Relates: https://github.com/cilium/cilium/pull/21112
Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
test/e2e: Remove l3 l4 related tests

This commit is to remove l3-l4 related tests, as these are
covered in connectivity tests:

https://github.com/cilium/cilium-cli/tree/master/connectivity/manifests

- client-egress-to-echo-service-account-deny.yaml
- echo-ingress-l7-http.yaml
- client-egress-to-echo.yaml
- client-egress-to-echo-service-account.yaml
- client-egress-to-echo-named-port-deny.yaml

Relates: https://github.com/cilium/cilium-cli/pull/1028
Relates: https://github.com/cilium/cilium/pull/21112
Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
test/e2e: Remove k8s default denied network polices

The network policy translation logic is already covered extensively as
part of unit tests, connectivity tests for equivalent CNP polices are
covered as well. Hence, these k8s network policy tests provides neither
extra coverage nor value.

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
test/e2e: Remove k8s default allowed network polices

The network policy translation logic is already covered extensively as
part of unit tests, connectivity tests for equivalent CNP polices are
covered as well. Hence, these k8s network policy tests provides neither
extra coverage nor value.

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
test/e2e: Remove entity related connectivity test

This commit is to remove entity related connectivity test, as this test is already
covered in connectivity tests:

https://github.com/cilium/cilium-cli/tree/master/connectivity/manifests

- deny-all-entities.yaml
- allow-host-entity.yaml
- allow-cluster-entity.yaml

Relates: https://github.com/cilium/cilium-cli/pull/1066
Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
test/e2e: Remove CNP update test

This test is to check the scenario if the same policy name can be
updated with new rules successfully, which should be covered in
unit tests (e.g. merging rules), not to mention that we should
not test k8s watcher mechanism.

Also, remove l7 related test in this Context, as the updated policies
(e.g. cnp-update*.yaml) didn't have anything related to kafka.

Signed-off-by: Tam Mach <tam.mach@cilium.io>

tklauser commented on test/k8s/net_policies.go:
```diff
@@ -80,34 +59,13 @@ var _ = SkipDescribeIf(func() bool {
 
 		demoPath = helpers.ManifestGet(kubectl.BasePath(), "demo-named-port.yaml")
 		l3Policy = helpers.ManifestGet(kubectl.BasePath(), "l3-l4-policy.yaml")
-		l3PolicyDeny = helpers.ManifestGet(kubectl.BasePath(), "l3-l4-policy-deny.yaml")
-		l3NamedPortPolicy = helpers.ManifestGet(kubectl.BasePath(), "l3-l4-policy-named-port.yaml")
-		l3NamedPortPolicyDeny = helpers.ManifestGet(kubectl.BasePath(), "l3-l4-policy-named-port-deny.yaml")
-		l7Policy = helpers.ManifestGet(kubectl.BasePath(), "l7-policy.yaml")
-		l7NamedPortPolicy = helpers.ManifestGet(kubectl.BasePath(), "l7-policy-named-port.yaml")
-		l7PolicyKafka = helpers.ManifestGet(kubectl.BasePath(), "l7-policy-kafka.yaml")
 		l7PolicyTLS = helpers.ManifestGet(kubectl.BasePath(), "l7-policy-TLS.yaml")
-		ICMPPolicy = helpers.ManifestGet(kubectl.BasePath(), "icmp-policy.yaml")
-		ICMPPolicyDeny = helpers.ManifestGet(kubectl.BasePath(), "icmp-policy-deny.yaml")
-		ICMPv6Policy = helpers.ManifestGet(kubectl.BasePath(), "icmp6-policy.yaml")
-		ICMPv6PolicyDeny = helpers.ManifestGet(kubectl.BasePath(), "icmp6-policy-deny.yaml")
 		TLSCaCerts = helpers.ManifestGet(kubectl.BasePath(), "testCA.crt")
 		TLSArtiiCrt = helpers.ManifestGet(kubectl.BasePath(), "internal-artii.crt")
 		TLSArtiiKey = helpers.ManifestGet(kubectl.BasePath(), "internal-artii.key")
 		TLSLyftCrt = helpers.ManifestGet(kubectl.BasePath(), "internal-lyft.crt")
 		TLSLyftKey = helpers.ManifestGet(kubectl.BasePath(), "internal-lyft.key")
 		TLSCa = helpers.ManifestGet(kubectl.BasePath(), "ca.crt")
-		serviceAccountPolicy = helpers.ManifestGet(kubectl.BasePath(), "service-account.yaml")
-		serviceAccountPolicyDeny = helpers.ManifestGet(kubectl.BasePath(), "service-account-deny.yaml")
-		knpDenyIngress = helpers.ManifestGet(kubectl.BasePath(), "knp-default-deny-ingress.yaml")
-		knpDenyEgress = helpers.ManifestGet(kubectl.BasePath(), "knp-default-deny-egress.yaml")
-		knpDenyIngressEgress = helpers.ManifestGet(kubectl.BasePath(), "knp-default-deny-ingress-egress.yaml")
-		cnpDenyIngress = helpers.ManifestGet(kubectl.BasePath(), "cnp-default-deny-ingress.yaml")
-		cnpDenyEgress = helpers.ManifestGet(kubectl.BasePath(), "cnp-default-deny-egress.yaml")
-		knpAllowIngress = helpers.ManifestGet(kubectl.BasePath(), "knp-default-allow-ingress.yaml")
-		knpAllowEgress = helpers.ManifestGet(kubectl.BasePath(), "knp-default-allow-egress.yaml")
-		cnpMatchExpression = helpers.ManifestGet(kubectl.BasePath(), "cnp-matchexpressions.yaml")
-		cnpMatchExpressionDeny = helpers.ManifestGet(kubectl.BasePath(), "cnp-matchexpressions-deny.yaml")
```
Just one note because I just opened a PR to remove some unused manifests: Please check whether these are in use by any other tests and if not, remove them as part of this PR as well.

sayboras replied:
good point, I just noticed that from your other PR :100: :medal_military:

tklauser commented on test/k8s/manifests/l7-policy.yaml:
```diff
@@ -1,21 +0,0 @@
-apiVersion: "cilium.io/v2"
```
This policy still seems to be used here: https://github.com/cilium/cilium/blob/062549799357504bb13ceb1e7854aef56ecf6130/test/k8s/updates.go#L63

sayboras replied:
thanks thanks, I will wait for test-1.25-net-next to correct all mistakes :100:

tklauser commented on test/k8s/manifests/cnp-default-deny-ingress.yaml:
```diff
@@ -1,8 +0,0 @@
-apiVersion: "cilium.io/v2"
```
This one is still in use here: https://github.com/cilium/cilium/blob/062549799357504bb13ceb1e7854aef56ecf6130/test/k8s/net_policies.go#L1585 and here: https://github.com/cilium/cilium/blob/062549799357504bb13ceb1e7854aef56ecf6130/test/k8s/net_policies.go#L1601

tklauser reviewed (changes requested):
Some of the deleted manifests still seem to be in use by other tests. I've added comments for some but there might be others. Would be worth to `grep` for all of the removed ones across the tree to check whether they are safe to delete.

sayboras committed:
test/e2e: Remove default ingress and egress deny related tests

This commit is to remove default CNP ingress, egress denied related tests, as these are
covered in connectivity tests:

https://github.com/cilium/cilium-cli/tree/master/connectivity/manifests

- deny-all-ingress.yaml
- deny-all-egress.yaml

Relates: https://github.com/cilium/cilium-cli/pull/1066
Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
test/e2e: Remove toEntities related tests

This commit is to remove toEntities related tests, as these are
covered in connectivity tests:

https://github.com/cilium/cilium-cli/tree/master/connectivity/manifests

- client-egress-to-entities-world.yaml
- allow-all-except-world.yaml

Relates: https://github.com/cilium/cilium-cli/pull/1028
Relates: https://github.com/cilium/cilium/pull/21112
Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
test/e2e: Remove l7 connectivity test

This commit is to remove L7 connecivity test, as this test is already
covered in connectivity tests:

https://github.com/cilium/cilium-cli/tree/master/connectivity/manifests

- echo-ingress-l7-http.yaml
- echo-ingress-l7-http-named-port.yaml

Relates: https://github.com/cilium/cilium-cli/pull/1066
Signed-off-by: Tam Mach <tam.mach@cilium.io>

tklauser reviewed (approved):
LGTM, approved contingent on CI tests passing.
